src/bpftrace.cpp

```diff
@@ -1363,6 +1432,49 @@ void BPFtrace::poll_perf_events(bool drain)
   return;
 }
 
+int BPFtrace::poll_perf_events()
+{
+  auto events = std::vector<struct epoll_event>(online_cpus_);
+  int ready = epoll_wait(epollfd_, events.data(), online_cpus_, timeout_ms);
+  if (ready <= 0)
+  {
+    return ready;
+  }
+  for (int i = 0; i < ready; i++)
+  {
+    perf_reader_event_read((perf_reader *)events[i].data.ptr);
+  }
+  return ready;
+}
+
+int BPFtrace::reset_rb_loss_counter()
+{
+  // reset ringbuf loss counter
+  return bpf_update_elem(
+      maps[MapManager::Type::RingbufLossCounter].value()->mapfd_,
+      const_cast<uint32_t *>(&rb_loss_cnt_key_),
+      const_cast<uint64_t *>(&rb_loss_cnt_val_),
+      0);
+}
+
+void BPFtrace::read_rb_loss_counter()
+{
+  uint64_t value = 0;
+  if (bpf_lookup_elem(
+          maps[MapManager::Type::RingbufLossCounter].value()->mapfd_,
+          const_cast<uint32_t *>(&rb_loss_cnt_key_),
+          &value))
+  {
+    LOG(ERROR) << "fail to get ringbuf loss counter";
+  }
+  if (value)
+  {
+    if (reset_rb_loss_counter())
+      LOG(ERROR) << "fail to reset ringbuf loss counter";
+    out_->lost_events(value);
```

Comment:
Seems it's possible to lose loss counts here (prog increments counter between when userspace reads and resets to 0). How about keeping track of the last read count and printing the delta each time? I think that should prevent races, right?